src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -1012,73 +1012,72 @@ private static unsafe void ZeroMatrixItemsCore(AlignedArray dst, int c, int ccol
         {
-            Contracts.AssertNonEmpty(src);
+            Contracts.AssertNonEmpty(source);
             Contracts.AssertNonEmpty(v);
             Contracts.AssertNonEmpty(w);
             Contracts.Assert(length > 0);
-            Contracts.Assert(length <= src.Length);
+            Contracts.Assert(length <= source.Length);
             Contracts.Assert(length <= v.Length);
             Contracts.Assert(length <= w.Length);
 
-            SdcaL1UpdateDense(primalUpdate, new Span<float>(src, 0, length), threshold, new Span<float>(v, 0, length), new Span<float>(w, 0, length));
+            SdcaL1UpdateDense(primalUpdate, new Span<float>(source, 0, length), threshold, new Span<float>(v, 0, length), new Span<float>(w, 0, length));
         }
 
-        private static void SdcaL1UpdateDense(float primalUpdate, Span<float> src, float threshold, Span<float> v, Span<float> w)
+        private static void SdcaL1UpdateDense(float primalUpdate, Span<float> source, float threshold, Span<float> v, Span<float> w)
         {
             if (Avx.IsSupported)
             {
-                AvxIntrinsics.SdcaL1UpdateU(primalUpdate, src, threshold, v, w);
+                AvxIntrinsics.SdcaL1UpdateU(primalUpdate, source, threshold, v, w);
             }
             else if (Sse.IsSupported)
             {
-                SseIntrinsics.SdcaL1UpdateU(primalUpdate, src, threshold, v, w);
+                SseIntrinsics.SdcaL1UpdateU(primalUpdate, source, threshold, v, w);
             }
             else
             {
-                for (int i = 0; i < src.Length; i++)
+                for (int i = 0; i < source.Length; i++)
                 {
-                    v[i] += src[i] * primalUpdate;
+                    v[i] += source[i] * primalUpdate;
                     float value = v[i];
                     w[i] = Math.Abs(value) > threshold ? (value > 0 ? value - threshold : value + threshold) : 0;
                 }
             }
         }
 
         // REVIEW NEEDED: The second argument "length" is unused even in the existing code.
-        public static void SdcaL1UpdateSparse(float primalUpdate, int length, float[] src, int[] indices, int count, float threshold, float[] v, float[] w)
+        public static void SdcaL1UpdateSparse(float primalUpdate, int length, float[] source, int[] indices, int count, float threshold, float[] v, float[] w)
         {
-            Contracts.AssertNonEmpty(src);
+            Contracts.AssertNonEmpty(source);
             Contracts.AssertNonEmpty(indices);
             Contracts.AssertNonEmpty(v);
             Contracts.AssertNonEmpty(w);
             Contracts.Assert(count > 0);
-            Contracts.Assert(count <= src.Length);
+            Contracts.Assert(count <= source.Length);
             Contracts.Assert(count <= indices.Length);
-            Contracts.Assert(count < length);
```

Comment:
Since you removed `length` from Sse.cs, can you also remove `length` from this method and remove that `REVIEW NEEDED` comment as well?

Reply:
That should now all be removed :)

Reply:
I think these changes are good. What's a few more to make it all good/consistent?

Reply:
@eerhardt I may be wrong on this, but I saw that some tests have failed. Looking into it, I think it may be that the `length` parameter is getting passed into native code here:

https://github.com/dotnet/machinelearning/blob/2983312386878b0047a9be64e3b12c05e82c7932/src/Native/CpuMathNative/Sse.cpp#L2665

With the removal of the parameter, I'm wondering if it's producing incorrect results.

Reply:
@jwood803, I've merged a separate "refactor CpuMathUtils" change #1229 that conflicts with your change. In that change, I've completely fixed the SdcaL1Update signatures so they no longer include an unused `length` parameter, and are now consistent. When you resolve the conflicts, you should be able to ignore this comment now.